### PR TITLE
Fixing _bundle provider + Adding to the Magician

### PR DESCRIPTION
**Pull request by rambleraptor**
The _bundle provider does generation on the puppet-google + chef-google repos. These download all of the different submodules at once.

This fixes their generation + adds them to the Magician.

-----------------------------------------------------------------
# [all]

### CODE REVIEW

**Comment by rambleraptor** [comment]
@ndmckinley I'm still doing some tests on Magician (this is a larger CI change than usual), but can you sanity check?

**Comment by rambleraptor** [comment]
@ndmckinley please take a look! The magician works on this PR and it's ready to be merged in.

File: .ci/ci.yml.tmpl [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+  }
+%}
 {% set terraform_enabled = true %}
 {% set ansible_enabled = true %}
 {% macro names_as_list(repo, names) -%}
 {% for name in names %}
 build/{{repo}}/{{name}} 
 {%- endfor %}
 {% endmacro -%}
+{% macro submodule_name(repo, submodule_hashes, prefix) -%}
```
Comment (ndmckinley) [inline .ci/ci.yml.tmpl]: This looks confusing - if it can't be made pretty, can we comment it so it makes sense?

File: .ci/ci.yml.tmpl [hunk 2]
```diff
@@ -96,15 +111,15 @@ resources:
     - name: puppet-{{module}}-intermediate
       type: git-branch
       source:
-        uri: git@github.com:((github-account.username))/puppet-google-{{module}}.git
+        uri: git@github.com:((github-account.username))/{{submodule_name(module, puppet_custom_submodules, 'puppet-google-').strip()}}.git
         private_key: ((repo-key.private_key))
```
Comment (ndmckinley) [inline .ci/ci.yml.tmpl]: probably too verbose - can we simplify it?

File: .ci/ci.yml.tmpl (hunk 1 above)
Comment (rambleraptor) [reply .ci/ci.yml.tmpl]: Made this much simpler now that I'm not trying to generalize it.

File: .ci/ci.yml.tmpl (hunk 2 above)
Comment (rambleraptor) [reply .ci/ci.yml.tmpl]: Made this much simpler now that I'm not trying to generalize it.

**Comment by rambleraptor** [comment]
On the Magician side, this produces a no-op with Seven (my new readability changes produce the same CI.yaml file)

**Pull request merged**